src/utils/utils_log.c

```diff
@@ -0,0 +1,251 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#define _GNU_SOURCE 1
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#ifdef __APPLE__
+#include <pthread.h>
+#endif
+
+#include <stdarg.h>
+#include <string.h>
+#include <time.h>
+
+#include "utils_common.h"
+#include "utils_log.h"
+
+#define LOG_MAX 8192
+#define LOG_HEADER 256
+#define MAX_FILE_PATH 256
+#define MAX_ENV_LEN 2048
+
+typedef struct {
+    int timestamp;
+    int pid;
+    utils_log_level_t level;
+    utils_log_level_t flushLevel;
+    FILE *output;
+} utils_log_config_t;
+
+utils_log_config_t loggerConfig = {0, 0, LOG_ERROR, LOG_ERROR, NULL};
+
+static const char *level_to_str(utils_log_level_t l) {
+    switch (l) {
+    case LOG_DEBUG:
+        return "DEBUG";
+    case LOG_ERROR:
+        return "ERROR";
+    case LOG_INFO:
+        return "INFO";
+    case LOG_WARNING:
+        return "WARN";
+    default:
+        ASSERT(0);
+        return "";
+    }
+}
+
+static const char *match_env_var(const char *envVar, const char *prefix,
```

Comment:
Could you please move this function into utils_common? I think it would be useful for parsing other env vars as well (e.g. for customizing test behavior under qemu, etc.)

Reply:
Done